scripts/pre_commit_linter.py

```diff
@@ -562,12 +585,14 @@ def _get_all_filepaths(input_path, input_filenames):
             else:
                 invalid_filepaths.append(filename)
         if invalid_filepaths:
-            print ('The following file(s) do not exist: %s\n'
-                   'Exiting.' % invalid_filepaths)
+            print('The following file(s) do not exist: %s\n'
+                  'Exiting.' % invalid_filepaths)
             sys.exit(1)
         all_filepaths = valid_filepaths
     else:
         all_filepaths = _get_changed_filepaths()
+        all_filepaths = [
+            filename.decode('utf-8') for filename in all_filepaths]
```

Comment:
How can we make sure future contributors know to do this?

Reply:
I have moved this decoding part to the function `_get_changed_filepaths()` which was the parent function. This function returns a byte string(as it returns a subprocess output) so we need to explicitly convert it to unicode. Some functions do need to be modified to return unicode.

Reply:
Ack, sounds good. Thanks!